### PR TITLE
Update interface of IsValidCommittedSeal and InsertBlock by go-ibft's update

### PR DESCRIPTION
**Pull request by Kourin1996**
# Description

Fix the method interfaces of IBFTBackend (IsValidCommittedSeal, InsertBlock) based on go-ibft's update (https://github.com/0xPolygon/go-ibft/pull/24)

**Please run `go mod vendor` after https://github.com/0xPolygon/go-ibft/pull/24 is merged before this PR is merged**

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

Please complete this section if you ran manual tests for this functionality, otherwise delete it

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: consensus/ibft/verifier.go [hunk 1]
```diff
@@ -116,9 +118,12 @@ func (i *backendIBFT) IsValidProposalHash(proposal, hash []byte) bool {
...
 }
 
-func (i *backendIBFT) IsValidCommittedSeal(proposalHash, seal []byte) bool {
+func (i *backendIBFT) IsValidCommittedSeal(
+	proposalHash []byte,
+	committedSeal *messages.CommittedSeal,
+) bool {
 	//	seal was generated based on the proposal hash
-	validator, err := ecrecoverImpl(seal, wrapCommitHash(proposalHash))
+	validator, err := ecrecoverImpl(committedSeal.Signature, wrapCommitHash(proposalHash))
```
Comment (zivkovicmilos) [inline consensus/ibft/verifier.go]: Do we need an additional check here to make sure that `validator == committedSeal.Signer`?

**Review by zivkovicmilos — APPROVED** [review]
Looks good 💯 

Make sure to update this branch as soon as we merge out the PR on `go-ibft` 🙏

File: consensus/ibft/verifier.go (hunk 1 above)
Comment (Kourin1996) [reply consensus/ibft/verifier.go]: @zivkovicmilos I'm afraid I don't understand what you mean. The latest develop code check it. Do you think we can remove this check?

**Comment by zivkovicmilos** [comment]
@Kourin1996 

Make sure to also add the `CommittedSeal` creation in `BuildCommitMessage` on the backend

File: consensus/ibft/verifier.go (hunk 1 above)
Comment (Kourin1996) [reply consensus/ibft/verifier.go]: Added at b44c2748eae722d4ffad0408f640f94127246649

Comment (Kourin1996) [reply consensus/ibft/verifier.go]: Added at b44c2748eae722d4ffad0408f640f94127246649

**Review by dbrajovic — APPROVED** [review]

**Pull request merged**